### DISCUSSION TITLE
Command for printing-email-addresses

### DISCUSSION THREAD

**Discussion by hochleitner** [Digital-Media/HagenbergThesis#119] [Ideas]
Using `\url{me@example.com}` for typesetting an e-mail address does not produce a correct `mailto:me@example.com` link.

Using `\href` from the hyperref package is a bit cumbersome, especially when used in combination with `\url` so that the e-mail-addresses can break gracefully. Also, `\href` produces invalid results for the PDF metadata when used in the `\author` macro (where an e-mail address can usually be found).

I propose the following command to be added to our collection of style files:

```latex
\newcommand{\email}[1]{\texorpdfstring{\href{mailto:#1}{\nolinkurl{#1}}}{<#1>}}
```

This produces a valid `mailto:` link with the given argument and generates a valid PDF string (in < > brackets), should the command be used in the preamble.

What do you think?

**Comment by imagingbook** [comment]
Looks good! I suggest to place the definition in `hgb.sty`.

**Reply by hochleitner** [reply]
Done. I also added a short subsection about `\url{}` and `\email{}` in HgbThesisTutorial and added \email to the CTAN manual.
